src/data/data_fetcher.py

```diff
@@ -106,8 +106,10 @@ def _make_api_url(
 def _handle_two_year_transaction_period(
     two_year_transaction_period: Union[int, str]
 ) -> str:
-    if two_year_transaction_period.isnumeric():
-        if (int(two_year_transaction_period) % 2) == 0:
+    if str(two_year_transaction_period).isnumeric():
+        if int(two_year_transaction_period) % 2 != 0:
```

Comment:
At this point, we're all `int` until the end. I'd add a `two_year_transaction_period = int(two_year_transaction_period)` right above this to declutter calling that again and again